docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -887,97 +891,140 @@ def get_docx_variables( text ):
+    """For a given set of specific cases, transform a
+    PDF field name into a standardized object name
+    that will be the value for the attachment field."""
+    GATHER_CALL = '.gather()'
+    if docassemble_var in reserved_whole_words:
+        return docassemble_var
+
+    if docassemble_var in reserved_var_plurals:
+        return docassemble_var + GATHER_CALL
 
-  # Break up label into its parts
-  label_groups = get_reserved_label_parts(reserved_prefixes, label)
+    # Everything before the first period and everything from the first period to the end
+    var_parts = re.findall(r'([^.]*)(\..*)*', docassemble_var)
 
-  # If no matches to automateable labels were found,
-  # just use the label as it is
-  if (label_groups is None or label_groups[1] == ''):
-    return label  # return without 
-    
-  # With reserved words, we're always using an index
-  # of the plural version of the prefix of the label
-  prefix = label_groups[1]
-  # Turn any singluars into plurals, e.g. 'user' into 'users'
-  var_start = pluralize_base(reserved_pluralizers_map, prefix)
-  if prefix == label: # it's just a standalone, like "defendant"
-    return var_start # Return the pluralized standalone variable
-
-  digit = label_groups[2]
-  index = indexify(digit)
-
-  # If it's a numbered singluar reserved prefix, e.g. user3
-  if label == prefix + digit:
-    # Return the plural plus the index, e.g. users[2]
-    return var_start + index
+    # test for existence (empty strings result in a tuple)
+    if not var_parts[0]:
+        return False
+    # The prefix, ensuring no key or index
+    prefix = re.sub(r'\[.+\]', '', var_parts[0][0])
+    has_reserved_prefix = prefix in reserved_pluralizers_map.values()
 
-  suffix = label_groups[3]
-  # Avoid transforming arbitrary suffixes into attributes
-  if not suffix in reserved_suffixes_map:
-    return label  # return it as is
-  else:    
-    suffix_as_attribute = turn_any_suffix_into_an_attribute(reserved_suffixes_map, suffix)
-    whole_var = "".join([var_start, index, suffix_as_attribute])
-
-    # Has to happen after docket number has a bracket
-    if (should_be_stringified(whole_var)):
-      result = get_stringifiable_version(whole_var)
-    else: result = whole_var
-
-    return result
-  
-
-def is_reserved_docx_label(label, docx_only_suffixes = generator_constants.DOCX_ONLY_SUFFIXES, reserved_whole_words = generator_constants.RESERVED_WHOLE_WORDS, reserved_pluralizers_map = generator_constants.RESERVED_PLURALIZERS_MAP, reserved_suffixes_map=generator_constants.RESERVED_SUFFIXES_MAP):
+    if not has_reserved_prefix:
+        return docassemble_var
+
+    suffix = var_parts[0][1]
+    if not suffix:  # If only the prefix
+        return prefix + GATHER_CALL
+    # If the suffix is also reserved
+    # Regex for finding all exact matches of docx suffixes
+    docx_only_suffixes_regex = '^' + '$|^'.join(docx_only_suffixes) + '$'
```

Comment:
To make sure I'm getting this right, this assumes that anything that has DOCX suffix on that list is something that should be triggered with `.gather()`, correct? If so, we might in the future add a comment to that effect as that may change. Or we may want to give that global var a different name so that its use is clear and we don't have accidental side effects from changing the values in it. `docx_plurals_suffixes`?

Reply:
You're right, and I don't think I intended to make the assumption that 'reserved names' == 'plural, list based variables'. I'm okay with just removing this whole section then if you agree with that, I don't want to add additional semantics and names to existing variable lists.

Reply:
Actually, with the discussion in https://github.com/SuffolkLITLab/docassemble-assemblylinewizard/pull/143#discussion_r565380939, I don't think we need to distinguish between `.gather()` and `.gather(complete_attribute='...')` anymore, so this entire section is irrelevant. Sorry!